src/services/completions.ts

```diff
@@ -1494,6 +1516,16 @@ namespace ts.Completions {
             if (typeMembers && typeMembers.length > 0) {
                 // Add filtered items to the completion list
                 symbols = filterObjectMembersList(typeMembers, Debug.assertDefined(existingMembers));
+                if (existingMembers && existingMembers.length > 0) {

```

Comment:
Instead of doing this here, I would suggest you set the sort text while adding the symbol itself.

Reply:
@sheetalkamat, Thank you for your review.
> set the sort text while adding the symbol itself.

Can you give me more detail about this? Do you mean I should move these codes to somewhere other than here?

Reply:
Yes it goes in `filterObjectMembersList` where you check for the criteria which is if symbols exist in spread assignment and set `symbolToSortTextMap[getSymbolId(symbol)] to sort priority` Same goes for optional. The sort text should always come from that map and not any other logic later on.